plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerInstanceRuntimeInfo.java

```diff
@@ -58,45 +51,57 @@ public LocalDockerInstanceRuntimeInfo(@Assisted ContainerInfo containerInfo,
-                                           containerInternalHostname),
+                                             containerExternalHostname,
+                                             dockerNodeInternalHostname,
+                                             containerInternalHostname),
               internalHostnameWithPrecedence(dockerNodeInternalHostname,
-                                           containerInternalHostname),
+                                             containerInternalHostname,
+                                             containerInfo.getNetworkSettings(),
+                                             useInternalAddress),
               machineConfig,
               devMachineServers,
-              allMachinesServers);
+              allMachinesServers,
+              useInternalAddress);
     }
 
     private static String externalHostnameWithPrecedence(String externalHostnameProperty,
-                                                              String externalHostnameAssisted,
-                                                              String internalHostnameProperty,
-                                                              String internalHostnameAssisted) {
+                                                         String externalHostnameAssisted,
+                                                         String internalHostnameProperty,
+                                                         String internalHostnameAssisted) {
 
-        String externalHostnameEnvVar = System.getenv(CHE_DOCKER_MACHINE_HOST_EXTERNAL);
-        if (externalHostnameEnvVar != null) {
-            return externalHostnameEnvVar;
-        } else if (externalHostnameProperty != null) {
+        if (externalHostnameProperty != null) {
             return externalHostnameProperty;
         } else if (externalHostnameAssisted != null) {
             return externalHostnameAssisted;
         } else {
-            return internalHostnameWithPrecedence(internalHostnameProperty,internalHostnameAssisted);
+            return internalHostnameWithPrecedence(internalHostnameProperty,
+                                                  internalHostnameAssisted,
+                                                  null,
+                                                  false);
         }
     }
 
     private static String internalHostnameWithPrecedence(String internalHostnameProperty,
-                                                              String internalHostnameAssisted) {
+                                                         String internalHostnameAssisted,
+                                                         NetworkSettings networkSettings,
+                                                         boolean useInternalAddress) {
+
+        if (useInternalAddress) {
+            String containerHostName = null;
+            if (networkSettings != null) {
+                containerHostName = networkSettings.getIpAddress();
```

Comment:
Is this IP address the IP of the Che server container? What we need is instead the IP addresses of the workspace containers.